docs/src/tutorial/parallelization.rst

```diff
@@ -0,0 +1,319 @@
+            i = __pyx_t_2;
+            total = total + /* some indexing code */;
+            
+        }
+    }
+
+.. _parallel-block:
+    
+``parallel`` blocks
+===================
+
+Much less frequently used than ``prange`` is Cython's ``parallel``
+operator.  ``parallel`` generates a block of code that is run simultaneously
+on multiple threads at once.  Unlike ``prange``, however, work is
+not automatically divided between threads.
+
+Here we present three common uses for the ``parallel`` block:
+
+Stringing together prange blocks
+--------------------------------
+
+There is some overhead in entering and leaving a parallelized section.
+Therefore, if you have multiple parallel sections with small
+serial sections in between it can be more efficient to
+write one large parallel block.  Any small serial
+sections are duplicated, but the overhead is reduced.
+
+In the example below we do an in-place normalization of a vector.
+The first parallel loop calculates the norm, the second parallel
+loop applies the norm to the vector, and we avoid jumping in and out of serial
+code in between.
+
+.. tabs::
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/normalize.pyx
+        
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/normalize.py
+
+The C code is approximately:
+
+.. code-block:: C
+
+    #pragma omp parallel private(norm) reduction(+:total)
+    {
+        /* some calculations of array size... */
+        #pragma omp for firstprivate(i) lastprivate(i)
+        for (__pyx_t_2 = 0; __pyx_t_2 < __pyx_t_3; __pyx_t_2++){
+            /* ... */
+        }
+        norm = sqrt(total);
+        #pragma omp for firstprivate(i) lastprivate(i)
+        for (__pyx_t_2 = 0; __pyx_t_2 < __pyx_t_3; __pyx_t_2++){
+            /* ... */
+        }
+    }
+
+Allocating "scratch space" for each thread
+------------------------------------------
+
+Suppose that each thread requires a small amount of scratch space
+to work in.  They cannot share scratch space because that would
+lead to data races.  In this case the allocation and deallocation
+is done in a parallel section (so occurs on a per-thread basis)
+surrounding a loop which then uses the scratch space.
+
+Our example here uses C++ to find the median of each column in
+a 2D array (just a parallel version of ``numpy.median(x, axis=0)``).
+We must reorder each column to find the median of it, but don't want
+to modify the input array.  Therefore, we allocate a C++ vector per
+thread to use as scratch space, and work in that.  For efficiency
+the vector is allocated outside the ``prange`` loop.
+
+.. tabs::
+
+    .. group-tab:: Cython
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/median.pyx
+        
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/parallelization/median.py
+
```

Comment:
I think visually will be better to move comments from examples to separate note section. This will also reduce duplication.

```suggestion

.. note::

    Pure and classic syntax examples are not quite identical
    since pure Python syntax does not support C++ "new", so we allocate the
    scratch space slightly differently

```